From c19ff65b8d143d7974ec0dd568b8f5b5af2489d7 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Mon, 2 Jan 2017 16:32:56 -0500
Subject: [PATCH] ShulkerBox Dupe Prevention

This ensures that Shulker Boxes can never drop their contents twice, and
that the inventory is cleared incase it some how also got saved to the world.

diff --git a/src/main/java/net/minecraft/server/BlockShulkerBox.java b/src/main/java/net/minecraft/server/BlockShulkerBox.java
index ab0ece557..997ed795b 100644
--- a/src/main/java/net/minecraft/server/BlockShulkerBox.java
+++ b/src/main/java/net/minecraft/server/BlockShulkerBox.java
@@ -100,6 +100,7 @@ public class BlockShulkerBox extends BlockTileEntity {
                     }
 
                     a(world, blockposition, itemstack);
+                    tileentityshulkerbox.clear(); // Paper - This was intended to be called in Vanilla (is checked in the if statement above if has been called) - Fixes dupe issues
                 }
             }
             world.updateAdjacentComparators(blockposition, iblockdata.getBlock());
-- 
2.17.2 (Apple Git-113)

